sycl/include/sycl/ext/oneapi/matrix/matrix.hpp

```diff
@@ -25,3 +25,6 @@
 #include <sycl/ext/oneapi/matrix/matrix-jit.hpp>
 #include <sycl/ext/oneapi/matrix/static-query.hpp>
 #endif
+#if (SYCL_EXT_ONEAPI_MATRIX == 3)
```

Comment:
This implementation can also benefit from the static query we have as well. Besides that the query can give the user information about what the implementation support, it can also construct the matrices and make the sizes optional for the user.

We should probably add this to matrix-jit.hpp and fork to using the AOT tensorcore implementation based on some option (AOT for tensorcore).
I am asking this because we should have one place that has the interface to make maintaining the code easy but also, since this interface is experimental, we expect it will be changed (like the use argument you introduce).  We should make the interface in one place so we only have to modify it in only one place.

Reply:
Hi @dkhaldi , We would like to get this merged.  Could you clarify what you would like me to change?  Thanks.

Reply:
Sorry for late reply, I was thinking you can have these defined under the new test macro = 3 in the same file so they don't get redefined.
However,  I think it will be best if we merge these as separate files. Once we add the use argument, we can reiterate on this to merge both files. What do you think?

Reply:
OK sure, I think that keeping them separate is a good idea for now.